### PR TITLE
after completing all work related to the AWS SDK v2 resource migration (#770), we should remove any remaining v1 calls.

### PR DESCRIPTION
**Pull request by wakeful**
## Description

Fixes #770 - after completing all work related to the AWS SDK v2 resource migration (#770), we should remove any remaining v1 calls.

We are almost done with the [v2 migration](https://github.com/gruntwork-io/cloud-nuke/blob/master/v2_migration_report/output.md). This WIP branch uses only AWS SDK v2 for communication with AWS APIs.
This should be the final MR required to completely remove v1 from the codebase.

## TODOs

- [x] ~~merge the v2 versions of EC2 and Redshift~~
- [x] clean up the `util` package
- [x] remove the usage of the `awserr` package
- [x] remove the migration report (`v2_migration_report`)

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.
- [ ] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)
Removed AWS SDKv1

### Migration Guide
n/a

### CODE REVIEW

**Comment by james03160927** [comment]
Hi @wakeful, let me know when it's ready for final review. Thanks for all the contribution!

**Comment by wakeful** [comment]
FYI @james03160927 

The last thing I want to add is a test for the `util/error.go` file, just to ensure it works fine after removing the `awserr` package. It should be ready for review tomorrow.

**Comment by wakeful** [comment]
hey @james03160927 

This is now ready for review. I have tried to create logical commits for this change.

File: aws/resource_registry.go [hunk 1]
```diff
@@ -174,30 +173,13 @@ func toAwsResourcesPointer(resources []AwsResource) []*AwsResource {
...
-			logging.Debug(fmt.Sprintf(
-				"[aws sdk cfg] using deprecated aws sdk v1 for resource %s",
-				(*resource).ResourceName(),
-			))
-			(*resource).Init(session)
-		}
+		logging.Debug(fmt.Sprintf(
+			"[aws sdk cfg] using aws sdk v2 for resource %s",
+			(*resource).ResourceName(),
+		))
```
Comment (james03160927) [inline aws/resource_registry.go:181]: Is this debug message necessary even after we migrate fo sdkV2 for everything?

File: aws/resources/base_resource.go [hunk 2]
```diff
@@ -114,5 +109,5 @@ func (br *BaseAwsResource) IsNukable(identifier string) (bool, error) {
 }
 
 func (br *BaseAwsResource) IsUsingV2() bool {
```
Comment (james03160927) [inline aws/resources/base_resource.go:111]: Can we just get rid of this function now?

File: aws/resources/base_resource.go [hunk 3]
```diff
@@ -28,11 +27,7 @@ type BaseAwsResource struct {
 	cancel   context.CancelFunc
 }
 
-func (br *BaseAwsResource) Init(_ *session.Session) {
-	br.Nukables = make(map[string]error)
-}
-
-func (br *BaseAwsResource) InitV2(cfg awsgoV2.Config) {
+func (br *BaseAwsResource) InitV2(cfg aws.Config) {
```
Comment (james03160927) [inline aws/resources/base_resource.go:30]: We can replace this with Init (instead of having V2 as the suffix?

**Review by james03160927 — APPROVED** [review]
Just a few nit comments. Otherwise LGTM. Will trigger tests for this.

**Comment by wakeful** [comment]
> Just a few nit comments. Otherwise LGTM. Will trigger tests for this.

sorted!

**Review by james03160927 — APPROVED** [review]
LGTM

**Pull request merged**